### PR TITLE
Add more notes to prow/phony README

### PR DESCRIPTION
**Pull request by chaodaiG**

### CODE REVIEW

File: prow/cmd/phony/README.md [hunk 1]
```diff
@@ -13,7 +13,11 @@ go run prow/cmd/hook/main.go
 --config-path=config/prow/config.yaml
 --plugin-config=config/prow/plugins.yaml
 --hmac-secret-file=path/to/hmac
--github-token-path=path/to/github-token
+--github-token-path=path/to/github-token
+
+# Note:
+# --deck-url is required because --dry-run flag requires it
```
Comment (alvaroaleman) [inline prow/cmd/phony/README.md:19]: Are you sure? That flag is only used for the `github` client

Comment (chaodaiG) [reply prow/cmd/phony/README.md:19]: That's what is says when I didn't supply it. The error message might be wrong then

Comment (chaodaiG) [reply prow/cmd/phony/README.md:19]: Why does `github` client need it btw?

Comment (alvaroaleman) [reply prow/cmd/phony/README.md:19]: github requires it so we don't do mutating api calls (we just do nothing instead).

And I was looking at the wrong binary (deck), where we indeed don't require it

/lgtm
/approve

**Pull request merged**